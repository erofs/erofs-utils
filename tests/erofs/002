#!/bin/sh
# SPDX-License-Identifier: GPL-2.0+
#
# test short symlink (fast symlink)
#
seq=`basename $0`
seqres=$RESULT_DIR/$(echo $0 | awk '{print $((NF-1))"/"$NF}' FS="/")

# get standard environment, filters and checks
. "${srcdir}/common/rc"

cleanup()
{
	cd /
	rm -rf $tmp.*
}

_require_erofs

# remove previous $seqres.full before test
rm -f $seqres.full

# real QA test starts here
echo "QA output created by $seq"

if [ -z $SCRATCH_DEV ]; then
	SCRATCH_DEV=$tmp/erofs_$seq.img
	rm -f SCRATCH_DEV
fi

localdir="$tmp/$seq"
rm -rf $localdir
mkdir -p $localdir

# mkfs short symlink (fast symlink)
FNAME=$(printf %8s | tr " " "x")
ln -s $FNAME $localdir/symlink

_scratch_mkfs $localdir >> $seqres.full 2>&1 || _fail "failed to mkfs"
_scratch_mount 2>>$seqres.full

MD5='0b0cfc07fca81c956ab9181d8576f4a8'

realmd5=$(readlink $SCRATCH_MNT/symlink | tr -d '\n' | md5sum | cut -d' ' -f1)
[ "x$realmd5" = "x$MD5" ] || _fail "-->checkmd5 FAILED"

_scratch_unmount

echo Silence is golden
status=0
exit 0
