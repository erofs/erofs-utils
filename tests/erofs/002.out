QA output created by 002
Silence is golden
